web-basics-browser-security-model/activities/give-to-get/src/index.template.php

```diff
@@ -0,0 +1,17 @@
+<!doctype html>
+<html>
+	<head>
+		<title>This is the title of the webpage!</title>
+	</head>
+	<body>
+		<?php
+			$message='<p>You have to <strong>ask</strong> for the <strong>flag</strong> to <strong>get</strong> it!</p>';
+			if (isset($_GET['ask']) && !empty($_GET['ask'])) {
+				if ($_GET['ask'] == "flag") {
```

Comment:
```suggestion
				if ($_GET['ask'] === "flag") {
```

Reply:
To avoid these kind of unwanted vulnerabilities we can also use Flask with `python:3.8.10-slim` for this kind of challenges + it's also a smaller image than PHP
```bash
$ docker images             
REPOSITORY   TAG           IMAGE ID       CREATED         SIZE
python       3.8.10-slim   b29d7e54a204   2 years ago     107MB
php          7.2-apache    75d564ede52f   2 years ago     376MB
```

Reply:
It's true that by doing this the participants might get errors trying to access other routes/methods (which we can also fix). So it's your choice here, but if you stay with PHP at least use strict comparison